src/components/PeoplePage.tsx

```diff
@@ -1,29 +1,138 @@
+
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+// eslint-disable-next-line import/extensions
+import { PeoplePageTitle } from '../page/PeoplePageTitle';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { useSearchParams } from 'react-router-dom';
+import { QueryParams } from '../enum/queryParams.enum';
+import { Sex } from '../enum/sex.enum';
 
 export const PeoplePage = () => {
+  // #region states
+  const [loading, setLoading] = useState(true);
+  const [fetchPeopleError, setFetchPeopleError] = useState(false);
+  const [people, setPeople] = useState<Person[]>([]);
+  const [query, setQuery] = useState('');
+  const [showFilters, setShowFilters] = useState(false);
+  // #endregion
+
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const activeCenturies = searchParams.getAll(QueryParams.Centuries) || [];
+  const activeSex = searchParams.get(QueryParams.Sex) as Sex || '';
+  const activeQuery = searchParams.get(QueryParams.Query) || '';
+
+  const toggleCenturies = (currentCentury: number) => {
+    const tempCenturies = !activeCenturies.includes(String(currentCentury))
+      ? [...activeCenturies, `${currentCentury}`]
+      : activeCenturies.filter(century => century !== String(currentCentury));
+
+    return tempCenturies;
+  };
+
+  const normalizedQuery = query.trim().toLowerCase();
+  const filteredPeople = people.filter(person =>
+    person.name.toLowerCase().includes(normalizedQuery),
+  );
+
+  const filterByCentury = () => {
+    if (activeCenturies.length) {
+      return filteredPeople.filter(person =>
+        activeCenturies.includes(String(Math.floor(person.born / 100) + 1)),
+      );
+    } else {
+      return filteredPeople;
+    }
+  };
+
+  const handleFilterBySex = (peopleList: Person[], activeSex: Sex) => {
+    switch (activeSex) {
+      case Sex.Male:
+        return peopleList.filter(person => person.sex === Sex.Male);
+      case Sex.Female:
+        return peopleList.filter(person => person.sex === Sex.Female);
+      default:
+        return peopleList;
+    }
+  };
+
+  const filteredByCentury = filterByCentury();
+  const visiblePeople = handleFilterBySex(filteredByCentury, activeSex as Sex);
+
+  // Define the fetchPeople function outside of useEffect
+  const fetchPeople = async () => {
+    setFetchPeopleError(false);
+
+    try {
+      const response = await getPeople();
+      setPeople(response);
+      setShowFilters(true);
+    } catch (error) {
+      setFetchPeopleError(true);
+    } finally {
+      setLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    fetchPeople();
+  }, []);
+
+  // #region conditions
+  const fetchingErrorNotification = !loading && fetchPeopleError;
+  const noPeopleNotification = !loading && !fetchPeopleError && !people.length;
+  const showPeopleTable = !loading && !fetchPeopleError && !!people.length;
```

Comment:
If there is an error or an active loader - ur list length will be 0.
u don't need to constantly duplicate conditions
```suggestion
  const fetchingErrorNotification = fetchPeopleError;
  const noPeopleNotification = !people.length;
  const showPeopleTable = !!people.length;
```